### PR TITLE
Fixes remove_invalid_notes (renamed remove_notes_with_no_duration) + minor improvements

### LINKED ISSUE #39: `remove_invalid_notes` never removes anything unless `verbose=False`
See
https://github.com/YatingMusic/miditoolkit/blob/38cfa38e7525e9ceb69720e64e2bd49b3eff644a/miditoolkit/midi/containers.py#L287-L305

especially line 300; if `verbose=True`, the code will never reach the actual removal loop.

### PR DESCRIPTION
**Pull request by Natooz**
Fixes #39 + tests for the new method + brings minor improvements and updates (docstring, type hints, actions versions)
@akx

### CODE REVIEW

File: miditoolkit/midi/containers.py [hunk 1]
```diff
@@ -284,26 +284,18 @@ def __init__(
...
-
-        # Remove the notes found
-        for note in notes_to_delete:
-            self.notes.remove(note)
-        return False
+    def remove_notes_with_no_duration(self):
+        """Removes (inplace) notes whose end time is before or at their start time."""
+        for i in range(self.nb_notes - 1, -1, -1):
+            if self.notes[i].start >= self.notes[i].end:
+                del self.notes[i]
```
Comment (akx) [inline miditoolkit/midi/containers.py:291]: I'd keep the old name as a compatibility alias:

```suggestion
    def remove_invalid_notes(self, verbose: bool = True) -> None:
        warnings.warn("Call remove_notes_with_no_duration() instead.", DeprecationWarning)
        return self.remove_notes_with_no_duration()

    def remove_notes_with_no_duration(self) -> None:
        """Removes (inplace) notes whose end time is before or at their start time."""
        for i in range(self.nb_notes - 1, -1, -1):
            if self.notes[i].start >= self.notes[i].end:
                del self.notes[i]
```

File: miditoolkit/midi/containers.py [hunk 2]
```diff
@@ -284,26 +284,18 @@ def __init__(
...
+
+    @property
+    def nb_notes(self) -> int:
```
Comment (akx) [inline miditoolkit/midi/containers.py:294]: FWIW, _nb._ for "number" is a francophone contraction – I'd use `note_count`, `num_notes` or similar so it's more understandable for English speakers.

**Comment by Natooz** [comment]
(fixing what's broken)

File: miditoolkit/midi/containers.py (hunk 2 above)
Comment (Natooz) [reply miditoolkit/midi/containers.py:303]: Indeed, ty for pointing that out! Fixed in the next commit

**Pull request merged**